### DISCUSSION TITLE
Mac: App no longer launches

### DISCUSSION THREAD

**Discussion by jpage4500** [shannah/jdeploy#151] [Q&A] [answered]
I've been using jdeploy for a few months now and it's been going great. But yesterday I pushed an update to my app. Today, when I went to test it I saw the launcher screen briefly (assuming it was updating) but the app never opened. I tried to run it a few more times but nothing happens anymore.

I'm using a Mac M1

Is there some way to debug what's happening? I see some log files in ~/.jdeploy/ folder but nothing has been modified in a while.

**Comment by shannah** [comment]
On Mac, you can sometimes get some helpful details by launching the app from the command line.

```bash
cd ~/Applications/Myapp.app/Contents/MacOS
./Client4JLauncher
```

**Reply by shannah** [reply]
Also, can you share a link to the app so I can take a look?

**Reply by jpage4500** [reply]
thanks - that did help show the error:

```
~/Applications/Android Device Manager.app/Contents/MacOS$ ls
Client4JLauncher
~/Applications/Android Device Manager.app/Contents/MacOS$ ./Client4JLauncher
Finished updating JRE Base /Users/jpage/.jdeploy/jre-arm64
Finished updating JRE Base /Users/jpage/.jdeploy/jre-arm64
Finished updating JRE Base /Users/jpage/.jdeploy/jre-arm64
Error: Unable to access jarfile /Users/jpage/.jdeploy/packages/android-device-manager/24.11.07-2134/jdeploy-bundle/AndroidDeviceManager.jar
```

Here's the app: https://www.jdeploy.com/~android-device-manager

**Reply by shannah** [reply, marked as answer]
Ah.  Looks like it doesn't like the ".07" in the version.  It normalizes the version to "24.11.7-2134" in some cases, which leads to a mismatch.

I'll add this to the list to fix, but for now, create a new release without the leading zero.  E.g. "24.11.7-2134"

**Comment by shannah** [comment]
Created a ticket #152 .  Closing discussion
